containers/src/Kokkos_Vector.hpp

```diff
@@ -147,6 +147,85 @@ class vector : public DualView<Scalar*, LayoutLeft, Arg1Type> {
 
   void clear() { _size = 0; }
 
+  iterator insert(iterator it, const value_type& val) {
+    return insert(it, 1, val);
+  }
+
+  iterator insert(iterator it, size_type count, const value_type& val) {
+    if ((_size == 0) && (it == begin())) {
+      resize(count, val);
+      DV::sync_host();
```

Comment:
Shouldn't there also be a `DV::modify_host();` here?

Reply:
No resize will do that